### PR TITLE
adborroto/PG 62 convert to typescript

### PR DESCRIPTION
**Pull request by adborroto**
> **Note** - Since this is a public repository, make sure that we're not publishing private data in the code, commit comments, or this PR.

> **Note for reviewers** - Please add a 2nd reviewer if the PR affects more than 15 files or 100 lines (not counting
  `package-lock.json`), if it incurs significant risk, or if it is going through a 2nd review+fix cycle.

## 📚 Context/Description Behind The Change

- Remove `ava` and add `jest`
- Modify unit test to be compatible with jest
- Typescripify the package
- Modify README

## 🚨 Potential Risks & What To Monitor After Deployment

Verify that is compatible with previous versions. I bump up compatibility just in case. I will generate and alpha release to test the changes

## 🧑‍🔬 How Has This Been Tested?

- [ ] Prepare alpha release
- [x] Test the package with new changes 
- [x] Verify pipeline works

## 🚚 Release Plan

Test & Release

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -29,6 +57,24 @@ class PromisePool extends Emitter {
     this._errors = [];
   }
 
+  /**
+   * Get the number of concurrent functions that can be executed.
+   *
+   * @returns {number} The number of concurrent functions.
+   */
+  get numConcurrent(): number {
```
Comment (adborroto) [inline src/index.ts:65]: Added this mostly for testing purposes.

File: README.md [hunk 2]
```diff
@@ -61,19 +59,17 @@ or
 $ npm i @mixmaxhq/promise-pool
 ```
 
-Changelog
----------
+## Changelog
```
Comment (jsalvata) [inline README.md:62]: Should we update this? Or remove it altogether, if we'll be maintaining a CHANGELOG file?

**Review by jsalvata — APPROVED** [review]
LGTM

Left only a petty comment.

File: package.json [hunk 3]
```diff
@@ -1,52 +1,66 @@
...
+    },
+    "devDependencies": {
+        "@mixmaxhq/prettier-config": "^1.0.0",
+        "@mixmaxhq/semantic-release-config": "^2.0.3",
+        "@mixmaxhq/ts-config": "^1.2.1",
+        "@types/node": "^22.7.4",
+        "@types/jest": "^28.1.3",
+        "@typescript-eslint/eslint-plugin": "^4.33.0",
+        "@typescript-eslint/parser": "^4.33.0",
+        "babel-cli": "^6.26.0",
```
Comment (bradvogel) [inline package.json]: Babel can be removed?

File: README.md (hunk 2 above)
Comment (adborroto) [reply README.md:62]: I think we should yes. But I don't know how to publish this package. Should I do it manually ? 
@bradvogel  do you know?, last release was 5y ago.

Comment (bradvogel) [reply README.md:62]: Yup, the old school way :-/. This repo doesn't have semantic-release that auto-publishes.

You should be able to run `mixmax dev npm publish` (instructions [here](https://docs.google.com/document/d/1KALf-WokfvoNXxDOBruTrT6yki5hqfkj1f6OYYTlnEg/edit#bookmark=id.ht1hs081qy7p)) which runs [this script](https://github.com/mixmaxhq/mixmax/blob/78f64b927e37825f445452dd0622d7472a2b4cf0/cmds/dev-cmds/npm-cmds/scripts/publish).

File: package.json (hunk 3 above)
Comment (adborroto) [reply package.json]: Solved!

**Pull request merged**